jetty-client/src/test/java/org/eclipse/jetty/client/ConnectionPoolTest.java

```diff
@@ -316,7 +318,10 @@ public void resolve(String host, int port, Promise<List<InetSocketAddress>> prom
         assertEquals(1, destinations.size());
         HttpDestination destination = (HttpDestination)destinations.get(0);
         AbstractConnectionPool connectionPool = (AbstractConnectionPool)destination.getConnectionPool();
-        assertEquals(2, connectionPool.getConnectionCount());
+        if (DUPLEX_MAX_DURATION == factory)
+            assertThat(connectionPool.getConnectionCount(), lessThanOrEqualTo(2)); // The connections can expire upon release.
+        else
+            assertThat(connectionPool.getConnectionCount(), is(2));
```

Comment:
This is one fix, but there are many other places where `connectionPool.getConnectionCount()` is called and may therefore fail too.

What's the point of making the duration only 10 ms?

I ask because setting the duration at 1000 ms would probabilistically avoid races, at the cost of possibly making just one test a little slower.

Reply:
Yes, it looks like `testQueuedRequestsDontOpenTooManyConnections`, `testConnectionMaxUsage` and `testIdleTimeoutNoRequests` could suffer from the same flakeyness. I'm going to check that more closely and adapt them if needed or document why not.

The 10ms duration is very short exactly to ferret out genuine race conditions in the code under test. It's not a delay that tests are waiting for so increasing it to 1000ms would pretty much never trigger an expiration which would make this pool variation useless.